pkg/broker/binding_operations.go

```diff
@@ -65,23 +68,45 @@ func (b Broker) Bind(ctx context.Context, instanceID string, bindingID string, d
 	}
 
 	// Create a new Atlas database user from the generated definition.
-	_, err = client.CreateUser(*user)
+	resultingUser, err := client.GetUser(bindingID)
 	if err != nil {
-		b.logger.Errorw("Failed to create Atlas database user", "error", err, "instance_id", instanceID, "binding_id", bindingID)
-		err = atlasToAPIError(err)
-		return
+		// Create a new Atlas database user from the generated definition.
+		_, err = client.CreateUser(*user)
+		if err != nil {
+			b.logger.Errorw("Failed to create Atlas database user", "error", err, "instance_id", instanceID, "binding_id", bindingID)
+			err = atlasToAPIError(err)
+			return
+		}
+
+		b.logger.Infow("Successfully created Atlas database user", "instance_id", instanceID, "binding_id", bindingID)
+
+		return brokerapi.Binding{
+			Credentials: ConnectionDetails{
+				Username: bindingID,
+				Password: password,
+				URI:      cluster.SrvAddress,
+			},
+		}, nil
 	}
 
-	b.logger.Infow("Successfully created Atlas database user", "instance_id", instanceID, "binding_id", bindingID)
+	if compareBindings(user, resultingUser) {
+		return
```

Comment:
What will be the function returning in this case? err is nil and spec is Zero. What's the meaning of this?

Reply:
this will just return StatusCreated, code 201 that is. Temporary though.